### PR TITLE
Remove junit4, blacklist it on pom.xml, update all tests to use junit…

### LINKED ISSUE #343: Blacklist junit4 from dependencies and remove all junit4 style test assertions
**Description**

Blacklist junit4 from dependencies and remove all junit4 style test assertions
 
**Use case or motivation behind the feature request**

We don't want to use outdated junit libraries at all
```sh
[INFO] +- com.icegreen:greenmail-junit5:jar:1.6.13:test
[INFO] |     \- junit:junit:jar:4.13.2:test
```
 
**Related issues**

 
**Additional context**

### PR DESCRIPTION
**Pull request by StrongestNumber9**
…5 commands

Optimized imports for src/test/ by using IDEAs automatic tool as spotless do not remove duplicated imports (wildcard+exact)

Fixes #343

### CODE REVIEW

File: src/test/java/com/teragrep/pth10/syntaxErrorTest.java [hunk 1]
```diff
@@ -59,9 +57,9 @@ public void syntaxTest() {
         String e;
         final String q;
         q = "index = archive_memory ( host = \"localhost\"  Deny";
-        e = "failed to parse at line 1:49 due to missing PARENTHESIS_R at '<EOF>'";
-        Throwable exception = assertThrows(IllegalStateException.class, () -> utils.getQueryAnalysis(q));
-        assertEquals(e, exception.getMessage());
+        e = "Assertions.failed to parse at line 1:49 due to missing PARENTHESIS_R at '<EOF>'";
```
Comment (51-code) [inline src/test/java/com/teragrep/pth10/syntaxErrorTest.java:60]: Five of the expected exception messages have been accidentally changed here in SyntaxErrorTest.

File: src/test/java/com/teragrep/pth10/utils.java [hunk 2]
```diff
@@ -88,7 +91,7 @@ public void syntaxError(
                     RecognitionException e
             ) {
                 throw new IllegalStateException(
-                        "failed to parse at line " + line + ":" + charPositionInLine + " due to " + msg,
+                        "Assertions.failed to parse at line " + line + ":" + charPositionInLine + " due to " + msg,
```
Comment (51-code) [inline src/test/java/com/teragrep/pth10/utils.java:94]: Exception message changed here as well.

**Review by 51-code — CHANGES_REQUESTED** [review]
Added comments about exception messages, otherwise looks good.

File: src/test/java/com/teragrep/pth10/syntaxErrorTest.java (hunk 1 above)
Comment (StrongestNumber9) [reply src/test/java/com/teragrep/pth10/syntaxErrorTest.java]: Pushed a fix for all accidental failed -> Assertions.failed conversions

File: src/test/java/com/teragrep/pth10/utils.java (hunk 2 above)
Comment (StrongestNumber9) [reply src/test/java/com/teragrep/pth10/utils.java]: Pushed a fix for all accidental failed -> Assertions.failed conversions

**Review by 51-code — APPROVED** [review]

**Review by kortemik — APPROVED** [review]

**Pull request merged**